Automattic/wp-calypso
# invite-accept: drop 'fetchInvite' flux action

This is the first PR in a series which removes the remaining `lib/invites` flux stores.

## Changes proposed in this Pull Request

* drop `fetchInvite` flux action
* remove `invite-accept-validation` flux store
* use component state exclusively

Note: I chose using internal component state because data is used exclusively in that component, nothing else is subscribing to it.

#### Testing instructions

Go to `/people/invites/<site>` and create a new invitation (the details don't matter that much) for a test account of yours. Hit the link you should receive via email. Replace the `wordpress.com` part of the URL with the the calypso.live link from the comment below.

The form should look more or less like the screenshot below. This PR only touches the part where we fetch invite data so if you see something like that, this should be fine. If you're not logged in you should see a signup form. In no case you should see an error.

<img width="432" alt="Screenshot 2020-11-17 at 15 55 53" src="https://user-images.githubusercontent.com/9202899/99405822-77f77b80-28ed-11eb-8059-f4fa2c5ec386.png">

Now manipulate the invitation key in the URL and make sure you see an error like this:

<img width="433" alt="Screenshot 2020-11-17 at 15 58 22" src="https://user-images.githubusercontent.com/9202899/99406089-cb69c980-28ed-11eb-82f6-bddcadd678b3.png">



part of #24149
